spec/std/string_spec.cr

```diff
@@ -773,6 +773,24 @@ describe "String" do
     it { "hello\n\n\n\n".chomp("").should eq("hello\n\n\n\n") }
 
     it { "hello\r\n".chomp("\n").should eq("hello") }
+
+    it "pre-computes string size if possible" do
+      {"!hello!", "\u{1f602}hello\u{1f602}", "\xFEhello\xFF"}.each do |str|
+        {"", "\n", "\r", "\r\n"}.each do |newline|
+          x = str + newline
+          x.size # side-effect
```

Comment:
**thought:** This should technically not be necessary because the size is already known. `String#+` calculates it when its known for both operands and they're literal instances in this case.
So this call does not introduce any side effect because the result is already there.
If we want to ensure that we could assert `x.size_known? == true`.